### PR TITLE
Feature/update ignore filter message

### PR DESCRIPTION
**Pull request by jonathansamines**
When the **eslint@2** upgrade was made, the `filterIgnoredFileMessages` function was broken, since the new **eslint@2** [changed the message](https://github.com/eslint/eslint/blob/master/lib/cli-engine.js#L241) produced for the ignored files.

Maybe somehow related to the reported error on [ember-cli-eslint#44](https://github.com/jonathanKingston/ember-cli-eslint/issues/44).

### CODE REVIEW

**Comment by jonathanKingston** [comment]
Hey! Thanks for this, it's been a while since I wrote that so makes sense it was broken.

It looks like they have added the ability to ignore those by option now:
https://github.com/eslint/eslint/blob/a35f1274cc553789e7a83d8cf501c82be7b6b342/lib/cli-engine.js#L662

This should mean that: https://github.com/jonathanKingston/broccoli-lint-eslint/blob/03a6d3f4934b11f7ac2d90c663c822dd372cf16f/lib/index.js#L71 can be changed to have a default of `ignore: true` if the key isn't defined.

Either way there is only one line change here in this pull: https://github.com/jonathanKingston/broccoli-lint-eslint/pull/30/files if we are going with this option then this should be cleaned.

If possible a test would be brilliant!

**Comment by jonathansamines** [comment]
I guess the option is for either ignore the `.eslintignore` patterns or not. Since if the option is `false` then the ignoreResult won´t be created and the lint warning will be [created instead](https://github.com/eslint/eslint/blob/master/lib/cli-engine.js#L666). I'll try to add some tests.

**Comment by jonathanKingston** [comment]
Ah yeah, I think that was there before actually thanks!

**Comment by jonathansamines** [comment]
@jonathanKingston I just added tests cases. Let me know what do you think.

**Comment by BrianSipple** [comment]
@jonathansamines pinging for a heads-up about a few conflicts after merging #25.

**Comment by jonathansamines** [comment]
@BrianSipple thanks for pinging. I solved the merging conflicts.

File: lib/index.js [hunk 1]
```diff
@@ -51,10 +51,11 @@ function filterIgnoredFileMessages(errors) {
 function filterAllIgnoredFileMessages(result) {
   const resultOutput = result;
 
-  result.results.map((resultItem) => {
+  result.results.forEach((resultItem) => {
     resultItem.messages = filterIgnoredFileMessages(resultItem.messages);
     return resultItem;
```
Comment (BrianSipple) [inline lib/index.js]: It looks like this is just a leftover from using `map`, but with `forEach`, the `return` actually becomes unnecessary 😄.

Comment (jonathansamines) [inline lib/index.js]: Completely true, removed ;)

**Comment by BrianSipple** [comment]
@jonathansamines Thanks!

**Pull request merged**